pkg/cli/cli.go

```diff
@@ -143,6 +146,92 @@ func (c *CLIRunner) CreateACL(
 	return nil
 }
 
+func formatACLs(acls interface{}) string {
+	content, err := json.MarshalIndent(acls, "", "  ")
+	if err != nil {
+		log.Warnf("Error marshalling acls: %+v", err)
+		return "Error"
+	}
+
+	return string(content)
+}
+
+// DeleteACL deletes a single ACL.
+func (c *CLIRunner) DeleteACL(ctx context.Context, filter kafka.DeleteACLsFilter) error {
+	c.printer("Checking if ACL exists for filter:\n%+v", formatACLs(filter))
+	c.startSpinner()
+	// First check that ACL exists
+	getFilter := kafka.ACLFilter{
+		ResourceTypeFilter:        filter.ResourceTypeFilter,
+		ResourceNameFilter:        filter.ResourceNameFilter,
+		ResourcePatternTypeFilter: filter.ResourcePatternTypeFilter,
+		PrincipalFilter:           filter.PrincipalFilter,
+		HostFilter:                filter.HostFilter,
+		Operation:                 filter.Operation,
+		PermissionType:            filter.PermissionType,
+	}
+	clusterACLs, err := c.adminClient.GetACLs(ctx, getFilter)
+	c.stopSpinner()
+	if err != nil {
+		return fmt.Errorf("Error fetching ACL info: \n%+v", err)
+	}
+
+	if len(clusterACLs) == 0 {
+		return fmt.Errorf("No ACL matches filter:\n%+v", formatACLs(filter))
+	}
+
+	if len(clusterACLs) > 1 {
```

Comment:
Another option is to change the design to allow for multiple ACLs to be deleted at once. This may add additional risk